widgetsnbextension/src/manager.js

```diff
@@ -190,12 +188,18 @@ WidgetManager.set_state_callbacks = function (load_callback, save_callback, opti
 var url = [window.location.protocol, '//', window.location.host, window.location.pathname].join('');
 var key = 'widgets:' + url;
 WidgetManager.set_state_callbacks(function() {
-    if (localStorage[key]) {
-        return JSON.parse(localStorage[key]);
-    }
-    return {};
+    return Promise.resolve(Jupyter.notebook.metadata.ipywidgets.state);
```

Comment:
you need to handle the case where `ipywidgets` is not defined. (Should it rather be called widgets?)